api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -126,16 +133,19 @@ public OrderGroup saveOrderGroup(OrderGroup orderGroup) throws APIException {
 	public synchronized Order saveRetrospectiveOrder(Order order, OrderContext orderContext) {
 		return saveOrder(order, orderContext, true);
 	}
-	
-	private Order saveOrder(Order order, OrderContext orderContext, boolean isRetrospective) {
-		if (order.getOrderId() != null) {
-			throw new UnchangeableObjectException("Order.cannot.edit.existing");
-		}
+
+	private void failOnExistingOrder(Order order) {
 		if (order.getDateActivated() == null) {
 			order.setDateActivated(new Date());
 		}
-		boolean isDrugOrder = DrugOrder.class.isAssignableFrom(getActualType(order));
-		Concept concept = order.getConcept();
+	}
+
+	private void ensureDateActivatedIsSet(Order order) {
```

Comment:
`ensureDateActivatedIsSet` is for setting dateActivated

this should be clear from the method names